feat(customize): add account handling for imported issues

narrowizard opened the pull request:
- Create or update accounts for creator and assignee
- Link accounts to issues in the database
- Add account data to issues_output.csv
- Verify account data with snapshot_tables/accounts.csv

#8400

<!--
Licensed to the Apache Software Foundation (ASF) under one or more
contributor license agreements.  See the NOTICE file distributed with
this work for additional information regarding copyright ownership.
The ASF licenses this file to You under the Apache License, Version 2.0
(the "License"); you may not use this file except in compliance with
the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
-->
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [ ] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [ ] I have added relevant tests.
- [ ] I have added relevant documentation.
- [ ] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

<!--
Thanks for submitting a pull request!

We appreciate you spending the time to work on these changes.
Please fill out as many sections below as possible.
-->

### Summary
What does this PR do?

### Does this close any open issues?
Closes xx

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

narrowizard committed:
feat(customize): add account handling for imported issues

- Create or update accounts for creator and assignee
- Link accounts to issues in the database
- Add account data to issues_output.csv
- Verify account data with snapshot_tables/accounts.csv

#8400

d4x1 commented on backend/plugins/customize/service/service.go:
```diff
@@ -262,22 +263,53 @@ func (s *Service) importCSV(file io.ReadCloser, rawDataParams string, recordHand
 		var err errors.Error
 		var id string
-		if record["id"] == nil {
+		idValue, ok := record["id"]
```
I've seen soem duplicated codes like:
```
idValue, ok := record["id"]
if !ok || idValue == nil {
	return errors.Default.New("record without id")
}
id, ok := idValue.(string)
if !ok || id == "" {
	return errors.Default.New("invalid or empty id")
}
```
I think we can simplify a function.

narrowizard replied:
Fixed. has extracted to `getStringField` function and added test case to it.

d4x1 commented on backend/plugins/customize/service/service.go:
```diff
@@ -262,22 +263,53 @@ func (s *Service) importCSV(file io.ReadCloser, rawDataParams string, recordHand
+		return "", nil // Return empty ID if name is empty, no error needed here.
+	}
+	now := time.Now()
+	accountId := fmt.Sprintf("csv:CsvAccount:0:%s", accountName)
```
Is `0` in `csv:CsvAccount:0:%s` correct?

narrowizard replied:
CSV is not a real data source within the system; there are no plugins or connections specifically for CSV. This ID is a mock, designed to match the format of other entities, so all data obtained through the API (which is in CSV format) shares this single "connection".

narrowizard committed:
refactor(customize): extract getStringField function for better record handling

- Add getStringField function to extract and validate string fields from records
- Use getStringField to replace repetitive code in issueHandlerFactory
- Add unit tests for getStringField to ensure its correctness

#8400
